server/athenian/api/internal/repos.py

```diff
@@ -0,0 +1,51 @@
+from __future__ import annotations
+
+import logging
+from typing import Optional, Sequence
+
+from athenian.api import metadata
+from athenian.api.internal.prefixer import Prefixer, RepositoryName, RepositoryReference
+from athenian.api.internal.settings import LogicalRepositorySettings
+
+
+def dereference_db_repositories(
+    repos: list[tuple[int, Optional[str]]],
+    prefixer: Prefixer,
+    logical_settings: LogicalRepositorySettings,
+) -> tuple[RepositoryName, ...]:
+    """Dereference the repository IDs into a RepositoryName sequence."""
+    references = parse_db_repositories(repos)
+    resolved = []
+    deleted_logical = []
+    for name in prefixer.dereference_repositories(references):
+        if name.logical and not logical_settings.repo_exists(name):
+            deleted_logical.append(str(name))
+            continue
+        resolved.append(name)
+    if deleted_logical:
+        if deleted_logical:
+            log = logging.getLogger(f"{metadata.__package__}.dereference_db_repositories")
+            log.warning("ignoring deleted logical repositories: %s", deleted_logical)
+    return tuple(resolved)
+
+
+def parse_db_repositories(
+    val: Optional[list[tuple[int, str | None]]],
+) -> Optional[list[RepositoryReference]]:
+    """Parse the raw value in DB repositories JSON column as list of `RepositoryReference`."""
+    if val is None:
+        return val
+    return [
+        RepositoryReference("github.com", repo_id, logical_name or "")
+        for repo_id, logical_name in val
+    ]
```

Comment:
Well, we should plan changing all the places in the DB to include the "github.com" prefix. This will reduce the pain from gitlab. Similarly to the reposets.